sklearn/feature_selection/tests/test_multivariate_select.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+import itertools
+import numpy as np
+from scipy import stats, sparse
+import pytest
+from sklearn.feature_selection import MultivariateFeatureSelector,k_sample_test
+from sklearn.utils._testing import assert_array_equal
+from sklearn.datasets import make_classification
+
+def test_k_sample_test():
+    #Make sure k_sample_test produces test statistic as expected 
+    X, y = make_classification(
+        n_samples=200,
+        n_features=20,
+        n_informative=4,
+        n_redundant=1,
+        n_repeated=0,
+        n_classes=8,
+        n_clusters_per_class=1,
+        flip_y=0.0,
+        class_sep=10,
+        shuffle=False,
+        random_state=0,
+    )
+
+    F = k_sample_test(X,y)
+    assert (F >= -1)
+    assert (F <= 1)
+
+def test_sparse_support():
+    #Make sure sparse data is supported
+    X, y = make_classification(
+        n_samples=200,
+        n_features=20,
+        n_informative=4,
+        n_redundant=1,
+        n_repeated=0,
+        n_classes=8,
+        n_clusters_per_class=1,
+        flip_y=0.0,
+        class_sep=10,
+        shuffle=False,
+        random_state=0,
+    )
+    sequential_filter = MultivariateFeatureSelector(k = 4)
+    sequential_filter.fit(sparse.csr_matrix(X),y)
+    sequential_filter.transform(sparse.csr_matrix(X))
+    
+
+def test_multivariate_feature_selector():
+    #Make sure that selector selects informative features
+    X, y = make_classification(
+        n_samples=200,
+        n_features=20,
+        n_informative=4,
+        n_redundant=1,
+        n_repeated=0,
+        n_classes=8,
+        n_clusters_per_class=1,
+        flip_y=0.0,
+        class_sep=10,
+        shuffle=False,
+        random_state=0,
+    )
+    sequential_filter = MultivariateFeatureSelector(k = 5)
+    sequential_filter.fit(X,y)
+    support = sequential_filter.get_support()
+    gtruth = np.zeros(20)
+    gtruth[:5] = 1
+    assert_array_equal(support, gtruth)
+
+def test_invalid_sample_size():
+    #Test selector with invalid sample size
+    X = np.array([[10, 20], [20, 20], [20, 30]])
+    y = np.array([[1], [0], [0]])
+    with pytest.raises(ValueError):
+        MultivariateFeatureSelector(k=1).fit(X, y)
+
+def test_invalid_k():
+    #Test selector with invalid k input
+    X, y = make_classification(
+        n_samples=200,
+        n_features=20,
+        n_informative=4,
+        n_redundant=1,
+        n_repeated=0,
+        n_classes=8,
+        n_clusters_per_class=1,
+        flip_y=0.0,
+        class_sep=10,
+        shuffle=False,
+        random_state=0,
+    )
+    with pytest.raises(ValueError):
```

Comment:
when doing 3 tests like this and just modifying a single parameters, use `pytest.mark.parametrize`. Here is an [example](https://github.com/srecupe2/scikit-learn/blob/e42fb227b891593520525cf6cb6302f0edfc12eb/sklearn/feature_selection/tests/test_feature_select.py#L93)

Reply:
Commenting to note this as the issue that I am referring to.

Reply:
just add a `pytest.mark.parametrize` where the parameters are `k` and `ValueError` and the other is `k` and `TypeError`